Formula/openvino.rb

```diff
@@ -0,0 +1,188 @@
+    resource("ittapi").stage buildpath/"thirdparty/ittapi/ittapi"
+    resource("xbyak").stage buildpath/"thirdparty/xbyak"
+    resource("onnx").stage buildpath/"thirdparty/onnx/onnx"
+
+    if Hardware::CPU.arm?
+      resource("openvino_contrib").stage buildpath/"openvino_contrib"
+      resource("arm_compute").stage buildpath/"openvino_contrib/modules/arm_plugin/thirdparty/ComputeLibrary"
+    else
+      resource("onednn_cpu").stage buildpath/"src/plugins/intel_cpu/thirdparty/onednn"
+    end
+
+    cmake_args = std_cmake_args + %w[
+      -DCMAKE_OSX_DEPLOYMENT_TARGET=
+      -DENABLE_TBBBIND_2_5=OFF
+      -DENABLE_CPPLINT=OFF
+      -DENABLE_CLANG_FORMAT=OFF
+      -DENABLE_NCC_STYLE=OFF
+      -DENABLE_INTEL_GPU=OFF
+      -DENABLE_TEMPLATE=OFF
+      -DENABLE_INTEL_GNA=OFF
+      -DENABLE_INTEL_MYRIAD_COMMON=OFF
+      -DENABLE_PYTHON=OFF
+      -DENABLE_SAMPLES=OFF
+      -DENABLE_COMPILE_TOOL=OFF
+      -DCPACK_GENERATOR=BREW
+      -DENABLE_SYSTEM_PUGIXML=ON
+      -DENABLE_SYSTEM_TBB=ON
+      -DENABLE_SYSTEM_PROTOBUF=ON
+    ]
+
+    if Hardware::CPU.arm?
+      cmake_args += %W[
+        -DOPENVINO_EXTRA_MODULES=#{buildpath}/openvino_contrib/modules/arm_plugin
+      ]
+    end
+
+    system "cmake", "-S", buildpath.to_s, "-B", "#{buildpath}/openvino_build", *cmake_args
+    system "cmake", "--build", "#{buildpath}/openvino_build"
+
+    # install only required components
+
+    components = %w[core core_dev
+                    cpu gpu batch multi hetero
+                    ir onnx paddle pytorch tensorflow]
+    components.each { |comp| system "cmake", "--install", "#{buildpath}/openvino_build", "--component", comp }
+  end
+
+  test do
+    ov_version = shell_output(Formula["pkgconfig"].bin/"pkg-config --modversion openvino").strip
+    pkg_config_flags = Utils.safe_popen_read("pkg-config", "--cflags", "--libs", "openvino").chomp.split
+
+    unless build.head
+      if OS.linux?
+        lib_suffix = "so.#{ov_version}"
+      elsif OS.mac?
+        lib_suffix = "#{ov_version}.dylib"
+      end
+
+      if Hardware::CPU.arm?
+        assert_path_exists lib/"openvino-#{ov_version}/libopenvino_arm_cpu_plugin.so"
+      else
+        assert_path_exists lib/"openvino-#{ov_version}/libopenvino_intel_cpu_plugin.so"
+      end
+
+      assert_path_exists lib/"openvino-#{ov_version}/libopenvino_auto_plugin.so"
+      assert_path_exists lib/"openvino-#{ov_version}/libopenvino_hetero_plugin.so"
+      assert_path_exists lib/"openvino-#{ov_version}/libopenvino_auto_batch_plugin.so"
+    end
+
+    assert_path_exists lib/"libopenvino.#{lib_suffix}"
+    assert_path_exists lib/"libopenvino_c.#{lib_suffix}"
+
+    assert_path_exists lib/"libopenvino_ir_frontend.#{lib_suffix}"
+    assert_path_exists lib/"libopenvino_onnx_frontend.#{lib_suffix}"
+    assert_path_exists lib/"libopenvino_paddle_frontend.#{lib_suffix}"
+    assert_path_exists lib/"libopenvino_tensorflow_frontend.#{lib_suffix}"
```

Comment:
Checking if files exist isn't a great test because they can still be non-functional so the test below is better

Reply:
yes, but here I want to make sure they are located in expected places according to `-DCPACK_GENERATOR=BREW`. 

Do you think these checks are excessive and we need to keep only the tests below? (which actually checks the plugins can be loaded)

Reply:
removed the file path checks.